### PR TITLE
feat: customize Marker using Composable

### LINKED ISSUE #41: RFC: Define custom marker images with Composables
For cases where info windows aren't flexible enough, or the marker images need to be dynamic in some way, it'd be great to be able to use a `@Composable` to render the marker images.

With Views, this was possible by creating a view and using `View.drawToBitmap`, which should roughly also be possible with a ComposeView (with caveats, due to the use of software rendering).

### PR DESCRIPTION
**Pull request by mosmb**
- Create a Marker Composable that accepts a Composable content
- Update BasicMapActivity to show the Composable Marker
---

- [X] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #41  🦕

### CODE REVIEW

File: maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt [hunk 1]
```diff
@@ -190,6 +190,75 @@ public fun Marker(
...
+ * @param zIndex the z-index of the marker
+ * @param onClick a lambda invoked when the marker is clicked
+ * @param onInfoWindowClick a lambda invoked when the marker's info window is clicked
+ * @param onInfoWindowClose a lambda invoked when the marker's info window is closed
+ * @param onInfoWindowLongClick a lambda invoked when the marker's info window is long clicked
+ * @param content composable lambda expression used to customize the marker's content
+ */
+@Composable
+@GoogleMapComposable
+public fun Marker(
```
[the remark this answers is not in the document]
Comment (mosmb) [reply maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt:219]: 100% agree on this, there are already a lot of Composable overloading for `Marker`.
Also, it's not convenient to use and distinguish inside the IDE

File: maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt [hunk 2]
```diff
@@ -190,6 +190,75 @@ public fun Marker(
     )
 }
 
+/**
+ * A composable for a marker on the map.
```
[the remark this answers is not in the document]
Comment (mosmb) [reply maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt:194]: Will update it

File: maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mosmb) [reply maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt:219]: I was thinking about it and yes.
It makes sense as this is the thing people will tend to search for when using `maps-compose`

**Comment by mosmb** [comment]
@kikoso 
I updated based on what you suggest.
Feel free to suggest any other changes that you may think of :)

**Comment by mosmb** [comment]
@kikoso 
Feel free to tell me if you see other things to edit

**Comment by mosmb** [comment]
@kikoso 
Yep I added the API key as a secret in my repository, so I think it should work

**Comment by mosmb** [comment]
@kikoso Tests seem to fail because of the API key
I added a secret `ACTIONS_API_KEY ` to my rep but doesn't seem to work though

**Comment by mosmb** [comment]
@kikoso 
Any news for this PR to be able to be merged?

**Comment by mosmb** [comment]
@kikoso Ok! Thank you for the information!

**Review by wangela — CHANGES_REQUESTED** [review]
Thank you for contributing this highly requested feature, @mosmb! Although the github action can't access the secrets to run tests on GitHub, would you still please add tests for the new functionality to keep our code coverage even? We'll be able to merge this PR when that task is done.

**Comment by mosmb** [comment]
Thank you @wangela for your feedback! 
What kind of test do you expect? 
As I can see, there's no test for now for `Marker`, except one testing that a state must not be reused.
Do you expect to do the same test but with `MarkerComposable`?

**Review by wangela — APPROVED** [review]

**Comment by WonderCsabo** [comment]
@mosmb checking out your code, i stumbled upon [this](https://github.com/googlemaps/android-maps-compose/pull/355/files#diff-3f37360306c965afa0b67bbcce39c6d17ecb9db7fc16e6c6b8fb7a5a0bce32d9R47) line:

```kotlin
composeView.draw(fakeCanvas)
```

I was wondering, why is this needed?

**Comment by mosmb** [comment]
@WonderCsabo In some cases, the `ComposeView` will not trigger the draw of its node. So in order to force that and not having a `Composable` that is not being drawn, we draw an empty canvas into the `ComposeView` to force it to enter into the drawing phase.
It's kinda of a hack but it works.
In Compose `1.5.0`, they added some utility method to `Picture` to be able to draw `Composable` into bitmaps. But as this library is not based on this version of Compose yet, we stick with this hack

**Comment by WonderCsabo** [comment]
@mosmb thanks for the info. Good to know there will be an official method to do this. Now i see it in the [docs](https://developer.android.com/jetpack/compose/graphics/draw/modifiers#composable-to-bitmap).

**Pull request merged**